contracts/wallet/Proxy.sol

```diff
@@ -31,21 +32,22 @@ contract Proxy {
         implementation = _implementation;
     }
 
-    function() external payable {
+    fallback() external payable {
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            let target := sload(0)
+            calldatacopy(0, 0, calldatasize())
+            let result := delegatecall(gas(), target, 0, calldatasize(), 0, 0)
+            returndatacopy(0, 0, returndatasize())
+            switch result
+            case 0 {revert(0, returndatasize())}
+            default {return (0, returndatasize())}
+        }
+    }
 
-        if (msg.data.length == 0 && msg.value > 0) {
+    receive() external payable {
+        if (msg.value > 0) {
```

Comment:
I think we can safely remove this `if` . The idea was to not log 0 ETH transfer, but there is no real reason to do that and we are basically making an unnecessary check on every transfer.

Reply:
Ref https://github.com/argentlabs/argent-contracts/pull/114#discussion_r434496224